flink-test-utils-parent/flink-connector-test-utils/src/test/java/org/apache/flink/connector/testframe/container/FlinkImageBuilderTest.java

```diff
@@ -0,0 +1,160 @@
+                        .setBaseImage(settings.getBaseImage())
+                        .asTaskManager()
+                        .build();
+        assertThat(image.getDockerImageName()).isEqualTo("flink-configured-taskmanager");
+        try (final GenericContainer<?> container = new GenericContainer<>(image)) {
+            container
+                    .withStartupCheckStrategy(new OneShotStartupCheckStrategy())
+                    .withCommand("cat", "/opt/flink/conf/config.yaml")
+                    .start();
+            assertThat(container.getLogs()).contains("rpc:\n    address: jobmanager");
+        }
+    }
+
+    @Test
+    void testFromDistTaskManagerWithHostParameter() throws ImageBuildException {
+        final Configuration taskManagerConf = new Configuration();
+        taskManagerConf.addAll(conf);
+        taskManagerConf.set(TaskManagerOptions.HOST, "taskmanager1-host");
+        ImageFromDockerfile image =
+                new FlinkImageBuilder()
+                        .setFlinkDistPath(tempDir)
+                        .setTempDirectory(tempDir)
+                        .setConfiguration(taskManagerConf)
+                        .setLogProperties(settings.getLogProperties())
+                        .setBaseImage(settings.getBaseImage())
+                        .asTaskManager()
+                        .build();
+        assertThat(image.getDockerImageName()).isEqualTo("flink-configured-taskmanager");
+        try (final GenericContainer<?> container = new GenericContainer<>(image)) {
+            container
+                    .withStartupCheckStrategy(new OneShotStartupCheckStrategy())
+                    .withCommand("cat", "/opt/flink/conf/config.yaml")
+                    .start();
+            assertThat(container.getLogs()).contains("host: taskmanager1-host");
+        }
+    }
+
+    @Test
+    void testFromDistWithLog4jFile() throws ImageBuildException {
+        ImageFromDockerfile image =
+                new FlinkImageBuilder()
+                        .setFlinkDistPath(tempDir)
+                        .setTempDirectory(tempDir)
+                        .setConfiguration(conf)
+                        .setLogProperties(settings.getLogProperties())
+                        .setBaseImage(settings.getBaseImage())
+                        .asJobManager()
+                        .build();
+        try (final GenericContainer<?> container = new GenericContainer<>(image)) {
+            container
+                    .withStartupCheckStrategy(new OneShotStartupCheckStrategy())
+                    .withCommand("cat", "/opt/flink/conf/log4j-console.properties")
+                    .start();
+            System.out.println(container.getLogs());
+            assertThat(container.getLogs()).contains("appender.console.name=ConsoleAppender");
+        }
+    }
+
+    @Test
+    void testJobManagerWithBaseImage() throws ImageBuildException {
+        final FlinkContainersSettings settings =
```

Comment:
You already have `settings` and `conf` as a instance variables, maybe you can also make two of them (as instance variable) with explicit names?